salt/modules/virt.py

```diff
@@ -1726,7 +1739,66 @@ def init(
     Initialize a new vm
 
     :param name: name of the virtual machine to create
-    :param cpu: Number of virtual CPUs to assign to the virtual machine
+    :param cpu:
+        Number of virtual CPUs to assign to the virtual machine or a dictionary with detailed information to configure
+        cpu model and topology. The structure of the dictionary is documented in :ref:`init-cpu-def`.
+
+        .. code-block:: yaml
+
+             cpu:
+               placement: static
+               cpuset: 0-11
+               current: 5
+               maximum: 12
+               vcpus:
+                 0:
+                   enabled: 'yes'
+                   hotpluggable: 'no'
+                   order: 1
+                 1:
+                   enabled: 'no'
+                   hotpluggable: 'yes'
+               attrib:
+                 match: minimum
+                 mode: custom
+                 check: full
```

Comment:
I would move `match`, `mode` and `check` one level up to remove `attrib`

Reply:
got an idea already :)